controllers/mtls_reconciler.go

```diff
@@ -0,0 +1,211 @@
+			if item.GroupVersionKind().Kind == kuadrantv1beta1.AuthorinoGroupKind.Kind {
+				return item, true
+			}
+			return nil, false
+		})
+		// add label to authorino deployment {"sidecar.istio.io/inject":"true"}}}}}
+		aDeployment := &v12.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				// TODO can't be hardcoded, this is just one example
+				Name:      "authorino",
+				Namespace: aobjs[0].GetNamespace(),
+			},
+		}
+		err := r.GetResource(eventCtx, types.NamespacedName{
+			Namespace: aDeployment.Namespace,
+			Name:      aDeployment.Name,
+		}, aDeployment)
+		if err != nil {
+			return fmt.Errorf("could not get authorino deployment %w", err)
+		}
+		aDeploymentMutators := make([]reconcilers.DeploymentMutateFn, 0)
+		aDeploymentMutators = append(aDeploymentMutators, reconcilers.DeploymentTemplateLabelIstioInjectMutator)
+		err = r.ReconcileResource(eventCtx, &v12.Deployment{}, aDeployment, reconcilers.DeploymentMutator(aDeploymentMutators...))
+		if err != nil {
+			return fmt.Errorf("could not add label to authorino deployment %w", err)
+		}
+
+		// find a limitador object, then find and update the associated deployment
+		lobjs := lo.FilterMap(topology.Objects().Objects().Items(), func(item machinery.Object, _ int) (machinery.Object, bool) {
+			if item.GroupVersionKind().Kind == kuadrantv1beta1.LimitadorGroupKind.Kind {
+				return item, true
+			}
+			return nil, false
+		})
+		// add label to limitador deployment {"sidecar.istio.io/inject":"true"}}}}}
+		lDeployment := &v12.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				// TODO can't be hardcoded, this is just one example
+				Name:      "limitador-limitador",
+				Namespace: lobjs[0].GetNamespace(),
+			},
+		}
+		err = r.GetResource(eventCtx, types.NamespacedName{
+			Namespace: lDeployment.Namespace,
+			Name:      lDeployment.Name,
+		}, lDeployment)
+		if err != nil {
+			return fmt.Errorf("could not get authorino deployment %w", err)
+		}
+		lDeploymentMutators := make([]reconcilers.DeploymentMutateFn, 0)
+		lDeploymentMutators = append(lDeploymentMutators, reconcilers.DeploymentTemplateLabelIstioInjectMutator)
+		err = r.ReconcileResource(eventCtx, &v12.Deployment{}, lDeployment, reconcilers.DeploymentMutator(lDeploymentMutators...))
+		if err != nil {
+			return fmt.Errorf("could not add label to limitador deployment %w", err)
+		}
+
+		peerAuth := &istiosecurity.PeerAuthentication{
+			ObjectMeta: controllerruntime.ObjectMeta{
+				Name:      "default",
+				Namespace: kObj.Namespace,
+				Labels:    KuadrantManagedObjectLabels(),
+			},
+			Spec: securityv1beta1.PeerAuthentication{
+				Mtls: &securityv1beta1.PeerAuthentication_MutualTLS{
+					Mode: securityv1beta1.PeerAuthentication_MutualTLS_STRICT,
+				},
+			},
+		}
+
+		logger.Info("creating peer authentication resource", "status", "processing")
+
+		err = r.CreateResource(eventCtx, peerAuth)
```

Comment:
Consider calling `ReconcileResource` and passing a mutator. Feel free to pick [CreateOnlyMutator](https://github.com/laurafitzgerald/kuadrant-operator/blob/mtls/pkg/reconcilers/base_reconciler.go#L65-L67)

Reply:
Hi @eguzki If I wanted to to also have an update as well as a create, it is the case that I can add a set of mutator functions? I see the observability reconciler is using the createonly for Pod and Service monitors. Is there pros or cons to each approach? I guess it might be how likely we consider something other than kuadrant-operator to make a change that we need to reconcile against right? cc @david-martin